src/core/middlewares/util/util.go

```diff
@@ -111,22 +129,104 @@ type MfInfo struct {
 	Tag        string
 	Digest     string
 
-	// Exist is to index the existing of the manifest in DB. If false, it's an new image for uploading.
-	Exist bool
+	References []distribution.Descriptor
+	Descriptor distribution.Descriptor
+
+	// manifestExist is to index the existing of the manifest in DB by (repository, digest)
+	manifestExist     bool
+	manifestExistErr  error
+	manifestExistOnce sync.Once
+
+	// artifact the artifact indexed by (repository, tag) in DB
+	artifact     *models.Artifact
+	artifactErr  error
+	artifactOnce sync.Once
+
+	ExclusiveBlobs []*models.Blob
```

Comment:
// ExclusiveBlobs include the blobs that belong to the manifest only and exclude the blobs that shared by other manifests in the same repo(project/repository).

please gives an explanation about what is ExclusiveBlobs.

Reply:
Added